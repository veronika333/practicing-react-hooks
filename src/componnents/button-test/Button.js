import React from 'react';

const Button = ({label}) => {
    return ( 
        <button data-testid="button">
            {label}
        </button>
     );
}
 
export default Button;