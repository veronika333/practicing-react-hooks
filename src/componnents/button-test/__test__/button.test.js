import React from 'react';
import ReactDom from 'react-dom';
import Button from '../Button';
import {render, cleanup} from '@testing-library/react';
// import jest-dom/extend-expect;
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

afterEach(cleanup);

it('button element renders without crashing', () => {
    const div = document.createElement("div");
    ReactDom.render(
        <Button />,
        div
    )
})

it('renders button correctly', () => {
const {getByTestId} = render(<Button label="Test this button"></Button>)
expect(getByTestId('button')).toHaveTextContent('Test this button')
})

it('matches snapshot', () => {
  const tree = renderer.create(<Button label="Test this button"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
})