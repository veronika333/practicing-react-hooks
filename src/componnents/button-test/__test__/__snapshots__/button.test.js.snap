// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  data-testid="button"
>
  Test this button
</button>
`;
